spec/dummy/app/controllers/inertia_conditional_sharing_controller.rb

```diff
@@ -16,4 +18,16 @@ def show
       show_only_prop: 1,
     }
   end
+
+  def show_with_a_problem
+    render inertia: 'EmptyTestComponent', props: {
+      show_only_prop: 1,
+    }
+  end
+
+  protected
+
+  def conditionally_share_a_prop
+    self.class.inertia_share incorrectly_conditionally_shared_prop: 1
```

Comment:
genuine question: Why are we explicitly testing that this fails as opposed to just not testing it if it's undocumented behavior?

Reply:
When I first wrote a variation of this test, it didn't fail. It actually passed. Before #121 , this code would _not_ conditionally share that prop. Because the inertia_share store is a class instance variable, it would be included on all subsequent requests after the first time it ran.

It's not very elegant code, but I think the use case is valid. If I can share data per-controller, why not per action? I wrote the original spec just to document the behavior.

However, @ElMassimo 's refactor actually prevents this from occurring at all because it freezes the shared data structure once the controller class is loaded into memory. And that _is_ (part of) the intended behavior of his change. So, the spec now documents/ensures you can't even try the contrived workaround for per-action data sharing. To me, this is a nice guardrail that fails loudly, and since I was surprised by the new behavior when I added the spec back, I think it's worth keeping!